tests/test_type_alias_type.py

```diff
@@ -388,3 +389,96 @@ class Div(BaseModel):
             'type': 'literal_error',
         }
     ]
+
+
+@pytest.mark.skipif(
+    sys.version_info < (3, 12),
+    reason='type statement is available from 3.12',
+)
+@pytest.mark.parametrize(
+    'input_value,error',
+    [
+        ([1], None),
+        (2, 'Input should be an instance of Sequence'),
+        (['foo'], None),
+        ([True], None),
+        ([[]], None),
+        ([None], None),
+        ([[1, '']], None),
+        ([{'a': None}], None),
+        ([{'a': [{'foo': 'bar'}]}], None),
+        # 5 errors:
+        # - value.0.str,
+        # - value.0.int,
+        # - value.0.bool,
+        # - value.0.`list[nullable[union[str,int,bool,...,dict[str,...]]]]`,
+        # - value.0.`dict[str,...]`.2.[key]
+        ([{2: None}], '5 validation errors for MyModel'),
+    ],
+    ids=repr,
+)
+def test_type_alias_with_type_statement(create_module, input_value, error):
+    module = create_module(
+        """
+from typing import Sequence
+from pydantic import BaseModel
+
+type JSON = str | int | bool | JSONSeq | JSONObj | None | JSONAlias
+type JSONObj = dict[str, JSON]
+type JSONSeq = list[JSON]
+JSONAlias = JSONSeq
```

Comment:
Can delete these as well?

Reply:
I would keep these as these cover recursive aliases (and the old way of type aliasing through assigning types.)